core/store/src/db/refcount.rs

```diff
@@ -51,6 +51,27 @@ pub fn decode_value_with_rc(bytes: &[u8]) -> (Option<&[u8]>, i64) {
     }
 }
 
+/// Strips refcount from an owned buffer.
+///
+/// Works like [`decode_value_with_rc`] but rather than returning reference to
+/// subslice of the argument, adjusts given argument by removing the refcount
+/// from it.  This also means that if the refcount is zero or negative, the
+/// argument will be cleared.
+pub(crate) fn strip_refcount(bytes: &mut Vec<u8>) -> i64 {
```

Comment:
This perhaps could be a more natural signature? 

```suggestion
pub(crate) fn strip_refcount(bytes: Vec<u8>) -> Vec<u8> {
```

Both call-sites do `<= 0` check anyway.